pkg/controller/core/workload_controller.go

```diff
@@ -543,7 +595,22 @@ func (r *WorkloadReconciler) reconcileCheckBasedEviction(ctx context.Context, wl
 	}
 	// at this point we know a Workload has at least one Retry AdmissionCheck
 	message := "At least one admission check is false"
-	if err := workload.Evict(ctx, r.client, r.recorder, wl, kueue.WorkloadEvictedByAdmissionCheck, message, "", r.clock); err != nil {
+	if err := workload.Evict(ctx, r.client, r.recorder, wl, kueue.WorkloadEvictedByAdmissionCheck, message, "", r.clock, workload.WithCustomPrepare(func() (*kueue.Workload, error) {
+		// Only set Requeued=false if there's a delayed retry configured
+		// Check if any admission check has RequeueAfterSeconds set
+		hasDelayedRetry := false
+		for i := range wl.Status.AdmissionChecks {
+			if wl.Status.AdmissionChecks[i].State == kueue.CheckStateRetry && wl.Status.AdmissionChecks[i].RequeueAfterSeconds != nil {
+				hasDelayedRetry = true
+				break
+			}
+		}
+		if hasDelayedRetry {
+			// Set Requeued=false so that when the backoff elapses, we can set it to true
+			workload.SetRequeuedCondition(wl, kueue.WorkloadEvictedByAdmissionCheck, "Evicted due to admission check retry", false)
+		}
+		return wl, nil
+	})); err != nil {
```

Comment:
Actually, this is getting complex, I'm no longer sure this is needed actually. 

It think it is enough to set Requeued only from job_controller as before on removing QuotaReservation, because it is only at this moment when we consider the workload as Pending anyway.

So, I would try to minimize the impact of the PR on pre-existing behavior actually if not needed.

Reply:
Yeah, I removed that code.